GVRf/Framework/src/org/gearvrf/FutureWrapper.java

```diff
@@ -0,0 +1,50 @@
+package org.gearvrf;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This is a wrapper to convert T to Future<T> so that we do not need many
+ * versions of constructor for {@link GVRSceneObject} for different combinations
+ * of {@link GVRMesh} or Future<{@link GVRMesh}>, {@link GVRTexture} or Future<
+ * {@link GVRTexture}>.
+ * 
+ * @param <V>
+ *            Internal type to be wrapped. It can be either mesh or texture.
+ */
+public class FutureWrapper<V> implements Future<V> {
+
+    private V value_ = null;
+
+    public FutureWrapper(V value) {
+        value_ = value;
+    }
+
+    @Override
+    public boolean cancel(boolean mayInterruptIfRunning) {
+        return false;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return false;
+    }
+
```

Comment:
This code is fine - for the current use cases. The only thing that gives me pause is that _some_ app will try to use it in a different context, and it will fail because `isCancelled()` and `isDone()` return the wrong values.

That is, if this checkin validates, I'll go ahead and merge it ... but I'd sure _prefer_ to see a Correct implementation of `Future` ...